### ISSUE TITLE
Wikipedia and Wikidata processing failed

### ISSUE THREAD

**Issue by kiril-me** [google/sling#283]
Hi,

I try to run SLING Wikipedia and Wikidata processing [https://github.com/google/sling/blob/caspar/doc/guide/wikiflow.md].

After running this script (./run.sh --build_wiki), I started getting errors.

As far as I understand what is wikidata (wikidata-20181001-all.json.bz2) format special characters. Can you point me is my thought correct? Or give me a hint where should I look to fix it. I am ready to contribute.

First run errors:
[2018-10-21 11:42:57.636205: E sling/frame/reader.cc:43] Error reading object line 1:1166: Unterminated string
[2018-10-21 11:42:57.636578: F sling/nlp/wiki/wikidata-importer.cc:90] Check failed: obj.IsFrame() �H�
[2018-10-21 11:42:57.636460: I sling/task/job.cc:395] Task wikidata/workers completed
[2018-10-21 11:42:57.637520: I sling/task/job.cc:395] Task wikidata/text-file-reader completed
./run.sh: line 3: 14805 Aborted                 (core dumped) python python/run.py $*

Second run errors:
2018-10-21 18:09:21.898652: I sling/task/job.cc:386] Channel 0 completed
[2018-10-21 18:09:21.898817: I sling/task/job.cc:395] Task wikidata/workers completed
[2018-10-21 18:09:21.898688: E sling/frame/reader.cc:43] Error reading object line 1:1166: Unterminated string
[2018-10-21 18:09:21.899786: I sling/task/job.cc:395] Task wikidata/text-file-reader completed
[2018-10-21 18:09:21.899883: F sling/nlp/wiki/wikidata-importer.cc:90] Check failed: obj.IsFrame() {"type":"it
em","id":"Q6697928","labels":{"en":{"language":"en","value":"Lucky Like That"},"de":{"language":"de","value":"
Lucky Like That"},"fr":{"language":"fr","value":"Lucky Like That"},"nl":{"language":"nl","value":"Lucky Like T
hat"}},"descriptions":{"nl":{"language":"nl","value":"lied van Clea"}},"aliases":{},"claims":{"P31":[{"mainsna
k":{"snaktype":"value","property":"P31","datavalue":{"value":{"entity-type":"item","numeric-id":7366,"id":"Q73
66"},"type":"wikibase-entityid"},"datatype":"wikibase-item"},"type":"statement","id":"Q6697928$853FE57D-5FC6-4
BF8-99AC-071A59FDB0DF","rank":"normal"}],"P175":[{"mainsnak":{"snaktype":"value","property":"P175","datavalue"
:{"value":{"entity-type":"item","numeric-id":3625982,"id":"Q3625982"},"type":"wikibase-entityid"},"datatype":"
wikibase-item"},"type":"statement","id":"Q6697928$8B883318-9B96-4284-BD3A-93A426646E2B","rank":"normal","refer
ences":[{"hash":"fa278ebfc458360e5aed63d5058cca83c46134f1","snaks":{"P143":[{"snaktype":"value","property":"P1
43","datavalue":{"value":{"entity-type":"item","numeric-id":328,"id":"Q328"},"type":"wikibase-entityid"},"data
type":"wikibase-item"}]},"snaks-order":["P143"]}]}],"
[2018-10-21 18:09:21.900160: I sling/task/job.cc:400] Task wikidata/text-file-reader done
./run.sh: line 3: 18236 Aborted                 (core dumped) python python/run.py $*

**Comment by anders-sandholm** [comment]
@kiril-me , thanks for trying out the SLING Wikipedia and Wikidata processing and for reporting the error.

I assume that you are using the latest version on the caspar branch and that you were able to download the dump files for both wikidata and (EN?) wikipedia without any errors? (./run.sh --download_wikidata --download_wikipedia) 

Also, just wondering what OS you are running on?

**Comment by kiril-me** [comment]
@anders-sandholm Yes, I used the casper version and downloaded the en wiki data. 

It might my wiki data were corrupted. Because I tried another setup (rahul1980:ontonotes) and I got till the parser wikipedia stage. I will retest with the latest casper and give you to know. 

I am using Ubuntu 16. What are that hardware requirements?

**Comment by anders-sandholm** [comment]
OK. Sounds good. You should be fine with Ubuntu 16. Main constraint for running the wiki processing is on local memory and storage. Hope it works out with the retest. (Might be worth to rename the local wikidata dump file - just to see if the new downloaded dump file is any different).

**Comment by ringgaard** [comment]
Could it be that your Wikidata dump `wikidata-20181001-all.json.bz2` is truncated? It should be 29542685400 bytes.

I would recommend having at least 200GB free disk space before starting (including tmp space) and 32 GB RAM.

**Comment by kiril-me** [comment]
I used 8 CPU and 30GB RAM Ubuntu machine everything works perfectly (./run.sh --build_wiki). It seems the first run data were corrupted (I redownloaded them during the day).

At the same time, I run overnight the same process +download file (./run.sh --download_wikidata --download_wikipedia --build_wiki). I used the 8 CPU and 32GB RAM CentOs box. The process was killed on the build_kb step. I might it was not a good idea to run the whole process in one run. So hopefully after rerun everything will works.
